### DISCUSSION TITLE
custom keymap/layouts

### DISCUSSION THREAD

**Discussion by ramnathk** [monkeytypegame/monkeytype#877] [Ideas] [answered]
It would be awesome if we can provide our own layout to learn rather than be restricted to the set available in the settings page. Ideally, if I can provide a json (qmk configurator's for example) and MT could identify the keymap and layout, that would be awesome.

If not allow me to pick from say split/non split keyboards and have some way to manually enter the key layout

Cheers and thanks for one the best typing tutor tools

**Comment by Miodec** [comment, marked as answer]
Planned #316, but not a priority.
